DiffMinPoly/src/solver_love_and_support.jl

```diff
@@ -61,26 +62,45 @@ function eliminate_with_love_and_support_modp(ode::ODE, x, p::Int, ord::Int=minp
     l = length(possible_supp)
     info && @info "The size of the estimates support is $(length(possible_supp))"
 
-    tim2 = @elapsed ls = build_matrix_multipoint(F, ode_mod_p, x_mod_p, ord, possible_supp, info = info)
+    tim2 = @elapsed M, pivot = build_matrix_multipoint(F, ode_mod_p, x_mod_p, ord, possible_supp, info = info)
                                                                     
-    info && @info "eval method $(tim2)"
+    info && @info "Building Linear System in: $(tim2)"
 
-    info && @info "linear system dims $(size(ls))"
+    strt = time()
+
+    new_order = find_new_order(M, pivot)
+    ls = reorder_ls(M, new_order, F)
+
+    info && @info "Linear System dims $(size(ls))"
+
+    new_supp = reorder_supp(possible_supp, new_order)
+
+    ls_UL, ls_LL, ls_LR = ls[1:pivot, 1:pivot], ls[(pivot + 1):l, 1:pivot], ls[(pivot + 1):l, (pivot + 1):l]
+    ls = nothing
+
+    info && @info "Linear System reduced in $(time() - strt)"
+
+    strt = time()
     
-    system_soltime = @elapsed ker = kernel(ls, side=:right)
-    info && @info "Linear system solved in $system_soltime"
+    v1 = kernel(ls_UL, side=:right)
+    temp_LL = -(ls_LL * v1)
+    v2 = solve(ls_LR, temp_LL, side=:right)
```

Comment:
I think in this line you implicitly assume that `temp_LL` is one-by-something, in other words, that the kernel of `ls_UL` is one-dimensional. This is not necessarily the case, for example, the code fails on this model 
```Julia
 @ODEmodel(
      x1'(t) = x1(t) + 3 * x1(t) * x2(t),
      x2'(t) = -5 * x2(t) + x1(t) * x2(t),
      y(t) = x1(t)
 )
```
This model was in the tests but it seems that you commented it out. I propose to uncomment all the models in the tests and make sure the code works for them.